src/seedu/addressbook/Main.java

```diff
@@ -105,12 +112,15 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) throws StorageOperationException {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
+            ui.showToUser(e.getMessage() + " - Storage file has read-only access.");
+            throw e;
```

Comment:
Consider the rationale behind throwing another exception. If the exception has been handled via showing error message to user in the previous line, is it still necessary to throw another exception?